src/components/Layout/DropdownMenu.astro

```diff
@@ -0,0 +1,122 @@
+---
+interface Props {
+  inlineMenu?: boolean;
+  label?: string;
+}
+
+// generate unique ids for each component so that we can connect the button to the dropdown without requiring consumer to provide an id
+// https://github.com/MicroWebStacks/astro-examples/blob/c278c3fc597bce013d627025a0c93758d72cc433/05_client-uid-counters/src/libs/utils.js#L8
+function uid() {
+  let date = (Date.now()).toString();
+  const sub = date.substring(date.length-6,date.length-1);
+  return sub+"_"+Math.floor(Math.random() * 10000)
+}
+
+let id = uid();
+---
+
+<div 
+  data-navigation-dropdown-menu 
+  data-unactivated
+  data-inline-menu={Astro.props.inlineMenu}
+> 
+  <button 
```

Comment:
`label` doesn't seem great because it provides a worse experience on other browsers if we don't add code to mark it as a button to assistive tech and handle key events in addition to clicks. I can't really find a recommendation anywhere to use `label`s to replace buttons in Safari, I'm not sure why DaisyUI chose it.

Reply:
I saw a separate website a few days back that also referenced that same bug but can't find it for now. Let's go with `button` for now but put a note with a link to that WebKit bug in code for future reference.